src/libraries/Microsoft.PowerFx.Connectors/Execution/FormulaValueSerializer.cs

```diff
@@ -0,0 +1,188 @@
+        internal static readonly DType DType_Record = new (DKind.Record);
+
+        internal string Serialize(OpenApiSchema schema, IEnumerable<NamedValue> fields)
+        {
+            StartSerialization(schema);
+            WriteObject(null, schema, fields);           
+            return GetResult();
+        }
+
+        private void WriteObject(string objectName, OpenApiSchema schema, IEnumerable<NamedValue> fields)
+        {
+            StartObject(objectName);
+
+            foreach (var property in schema.Properties)
+            {
+                var namedValue = fields.FirstOrDefault(nv => nv.Name.Equals(property.Key, StringComparison.OrdinalIgnoreCase));
+
+                if (namedValue == null)
+                {
+                    throw new NotImplementedException($"Missing property {property.Key}, object is too complex or not supported");
+                }
+
+                WriteProperty(property.Key, property.Value, namedValue.Value);
+            }
+
+            EndObject();
+        }
+
+        private void WriteProperty(string propertyName, OpenApiSchema propertySchema, FormulaValue fv)
+        {
+            switch (propertySchema.Type)
+            {
+                case "array":
+                    // array
+                    AssertType(DType_Table, fv, propertyName);
+                    StartArray(propertyName);
+                         
+                    foreach (var item in (fv as TableValue).Rows)
+                    {
+                        if (item.Value is not RecordValue rva)
+                        {
+                            throw new ArgumentException($"Invalid type in array, expecting a RecordValue for {propertyName}, row type is {item.GetType().FullName}");
+                        }
+
+                        if (rva.Fields.Count() != 1)
+                        {
+                            throw new ArgumentException($"Incompatible Table for supporting array, RecordValue has more than one column - propertyName {propertyName}, number of fields {rva.Fields.Count()}");
+                        }
+
+                        StartArrayElement(propertyName);
+                        WriteValue(rva.Fields.First().Value);
+                    }
+                    
+                    EndArray();
+                    break;
+
+                case "null":
+                    // nullable
+                    AssertType(DType.ObjNull, fv, propertyName);
+                    WritePropertyValue(propertyName, fv);
+                    break;
+
+                case "number":
+                    // float, double
+                    AssertType(DType.Number, fv, propertyName);
+                    WritePropertyValue(propertyName, fv);
+                    break;
+
+                case "boolean":
+                    AssertType(DType.Boolean, fv, propertyName);
+                    WritePropertyValue(propertyName, fv);
+                    break;
```

Comment:
case "boolean": 

if (value is BooleanValue booleanValue)
            {
                WriteBooleanValue(booleanValue.Value);                
            }  else {error!!!}